### ISSUE TITLE
UNC not working in glob.sync

### ISSUE THREAD

**Issue by lfk-cbrain** [isaacs/node-glob#582]
Hello! I have tried to look at the documentation and from previous issues (#74 #167) regarding this topic, but I simply cannot get it to work

glob version: v10.3.12, 

I'm trying to access files on UNC with glob.sync, like this:

```js
    validateConfiguration(settingsFile: string, stepsPathes: StepSettings, workSpaceRoot: string): Diagnostic[] {
        return stepsPathes.reduce((res, path) => {
            const files = glob.sync(path, {ignore: '.gitignore' });
            if (!files.length) {
                const searchTerm = path.replace(workSpaceRoot + '/', '');
                const range = getTextRange(workSpaceRoot + '/' + settingsFile, `"${searchTerm}"`);
                res.push({
                    severity: DiagnosticSeverity.Warning,
                    range: range,
                    message: `No steps files found`,
                    source: 'cucumberautocomplete'
                });
            }
            return res;
        }, []);
    }
``` 
where path is written like `"//scs2/F2/Sc2Cloud08.Test/Integrations/Cucumber/TestFramework/ScenarioSteps/*.cs"` 

I have tried to include cwd:"//scs2/", and playing around with changing the path.

By debugging, I can see it sets UNC to true in the Pattern object method on `glob.js:170` and the Pattern looks correct.
However it returns an empty array as the matches.

Would love if anyone could help me out :)

**Comment by isaacs** [comment]
What does this node program output when run?

```js
require("fs").readdirSync(
  "//scs2/F2/Sc2Cloud08.Test/Integrations/Cucumber/TestFramework/ScenarioSteps"
)
```

**Comment by isaacs** [comment]
Will reopen when/if requested info is provided.

**Comment by lfk-cbrain** [comment]
Hello @isaacs 
Sorry for late reply.

I get the following stack

`"Error [ERR_UNC_HOST_NOT_ALLOWED]: UNC host 'scs2' access is not allowed\n    at Object.readdirSync (node:fs:1516:10)\n    at Object.readdirSync (node:electron/js2c/node_init:2:11289)\n    at watchFiles (c:\\Projects\\ProcessWork\\VSCucumberAutoComplete\\gclient\\server\\server.js:59:27)\n    at c:\\Projects\\ProcessWork\\VSCucumberAutoComplete\\gclient\\server\\server.js:81:9\n    at handleNotification (c:\\Projects\\ProcessWork\\VSCucumberAutoComplete\\gclient\\server\\node_modules\\vscode-jsonrpc\\lib\\main.js:483:43)\n    at processMessageQueue (c:\\Projects\\ProcessWork\\VSCucumberAutoComplete\\gclient\\server\\node_modules\\vscode-jsonrpc\\lib\\main.js:255:17)\n    at Immediate._onImmediate (c:\\Projects\\ProcessWork\\VSCucumberAutoComplete\\gclient\\server\\node_modules\\vscode-jsonrpc\\lib\\main.js:242:13)\n    at process.processImmediate (node:internal/timers:476:21)\n    at process.callbackTrampoline (node:internal/async_hooks:130:17)"`

In short "UNC host 'scs2' access is not allowed"

It is a work server, so might be blocking me access. However, I can open files on the scs2 server in vscode and edit them, so it does not seem like that.

**Comment by isaacs** [comment]
If node can't read the directories in the share, then node-glob can't tell you which files are in the directories in the share, it's as simple as that. Figure out why node can't read those dirs, and you'll have your answer. Nothing else I can do from here.
